packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -3,20 +3,17 @@ import cdk = require('@aws-cdk/cdk');
 import { RepositoryRef } from './repository';
 
 /**
- * Construction properties of the {@link PipelineSource CodeCommit source CodePipeline Action}.
+ * Common properties for creating {@link PipelineSource} -
+ * either directly, through its constructor,
+ * or through {@link RepositoryRef#addToPipelineStage}.
  */
-export interface PipelineSourceProps extends codepipeline.CommonActionProps {
+export interface CommonPipelineSourceProps {
```

Comment:
Should this be in the `codepipeline` package and extend `CommonActionProps`?

Reply:
I don't think so - `CommonPipelineSourceProps` contains things like `branch` and `pollForSourceChanges`, which are CodeCommit-specific, so I don't think it belongs in the `codepipeline` package.